### DISCUSSION TITLE
store undefined problem on vuex

### DISCUSSION THREAD

**Discussion by Lalrinfela** [quasarframework/quasar#13035] [General - Components / Directives / etc]
Hi, 

I have a problem access the store on my quasar app (router/index.js), **it was working yesterday, this morning i just updated it and it breaks after update**. I thought it might be because of some outdated issue. Below is my updated changes.

<img width="415" alt="Screenshot 2022-04-02 at 9 04 53 AM" src="https://user-images.githubusercontent.com/1894559/161364489-67369eb4-d4e5-4e2a-9696-5519b55d257d.png">

**However, to test on a fresh copy, I created a new fresh quasar app project with same Vuex, the store is still undefined as well when console log in a clean fresh quasar app. Am I missing anything please?**

<img width="1290" alt="Screenshot 2022-04-02 at 8 56 57 AM" src="https://user-images.githubusercontent.com/1894559/161364315-643efd35-35d5-413b-9e45-ecb883b31ddc.png">

**Comment by rowild** [comment]
I checked my setup, which uses vite, updated to quasar 2.6.5 and it works.

![Bildschirmfoto 2022-04-02 um 08 29 28](https://user-images.githubusercontent.com/213803/161370106-fe54fdb8-ea48-421b-8f8a-1e69d0bd8bc3.png)

Did you already try to delete node_modules and install anew?

**Reply by Lalrinfela** [reply]
I am using a fresh copy with webpack and its not working. The reason for using webpack is that my existing project uses webpack. Thanks BTW.

**Reply by rowild** [reply]
I updated a webpack project from quasar 2.6.4 to 2.6.5 and @quasar/extras to v1.13.4, but I kept @quasar/app (updated it to 3.3.3., though [screenshot is from before update]). The project is still working.

Why did you switch from @quasar/app to @quasar/app-webpack? (I do not know the difference of those 2 packages.)

![Bildschirmfoto 2022-04-02 um 12 00 17](https://user-images.githubusercontent.com/213803/161378503-4d95f1d6-df65-4500-b4a0-826be57bb923.png)
)

**Reply by rowild** [reply]
Kept on testing: removed @quasar/app and added @quasar/app-webpack instead (all with yarn). The console.log in the router/index.js file is till working fine.

**Reply by rstoenescu** [reply]
@Lalrinfela we triple checked everything and it works correctly. If you still encounter an issue, please open up a Github ticket with a minimal reproduction repo and the output of your "quasar info".

@rowild Please follow the release notes. q/app has been renamed to q/app-webpack going forward, to better differentiate between the Webpack and Vite variants. q/app v3 will no longer receive updates -- only q/app-webpack which also continues the versioning from q/app (it's the same thing, just renamed). If you have latest "@quasar/cli" globally installed, you can run "quasar upgrade -i" and it'll take care of the upgrade itself in a more thorough way.

**Reply by Lalrinfela** [reply]
@rstoenescu As instructed, i tried to create a ticket, but the live codesandbox cannot reproduce the error either.

Anyway, I have put my quasar info below. One thing that strike me is that i have a vuex in my composer dependecies, but the quasar info output gives vuex as Not installed (BTW, this is a fresh installed copy) - could this be the problem?:

Operating System - Darwin(21.4.0) - darwin/x64
NodeJs - 17.8.0

Global packages
  NPM - 8.5.5
  yarn - 1.22.18
  @quasar/cli - 1.3.2
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.6.5 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-webpack - 3.4.7 -- Quasar Framework App CLI with Webpack
  @quasar/extras - 1.13.5 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.31 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.14
  pinia - Not installed
  vuex - Not installed
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.17.8 -- Babel compiler core.
  webpack - 5.71.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.7.4 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.5 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - Lalrinfelas-MacBook-Pro.local
  en0 - 192.168.1.2

**Reply by MilosPaunovic** [reply]
> @rstoenescu As instructed, i tried to create a ticket, but the live codesandbox cannot reproduce the error either.
> 
> Anyway, I have put my quasar info below. One thing that strike me is that i have a vuex in my composer dependecies, but the quasar info output gives vuex as Not installed (BTW, this is a fresh installed copy) - could this be the problem?:
> 
> Operating System - Darwin(21.4.0) - darwin/x64
> NodeJs - 17.8.0
> 
> Global packages
>   NPM - 8.5.5
>   yarn - 1.22.18
>   @quasar/cli - 1.3.2
>   @quasar/icongenie - Not installed
>   cordova - Not installed
> 
> Important local packages
>   quasar - 2.6.5 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
>   @quasar/app-webpack - 3.4.7 -- Quasar Framework App CLI with Webpack
>   @quasar/extras - 1.13.5 -- Quasar Framework fonts, icons and animations
>   eslint-plugin-quasar - Not installed
>   vue - 3.2.31 -- The progressive JavaScript framework for building modern web UI.
>   vue-router - 4.0.14
>   pinia - Not installed
>   vuex - Not installed
>   electron - Not installed
>   electron-packager - Not installed
>   electron-builder - Not installed
>   @babel/core - 7.17.8 -- Babel compiler core.
>   webpack - 5.71.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
>   webpack-dev-server - 4.7.4 -- Serves a webpack app. Updates the browser on changes.
>   workbox-webpack-plugin - Not installed
>   register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
>   typescript - 4.5.5 -- TypeScript is a language for application scale JavaScript development
>   @capacitor/core - Not installed
>   @capacitor/cli - Not installed
>   @capacitor/android - Not installed
>   @capacitor/ios - Not installed
> 
> Quasar App Extensions
>   *None installed*
> 
> Networking
>   Host - Lalrinfelas-MacBook-Pro.local
>   en0 - 192.168.1.2
> 
> 

Vuex does not come preinstalled anymore. You need to add it yourself. Seems like you haven't clicked space button to select it in project creation.

**Reply by Lalrinfela** [reply]
@MilosPaunovic It was checked, thats why it is listed in packages.json. I even manually installed it again thinking it might be the reason.
<img width="846" alt="Screenshot 2022-04-02 at 10 54 56 PM" src="https://user-images.githubusercontent.com/1894559/161394362-19769a7d-1d30-4b04-bc4a-2a3b135ec6de.png">

**Comment by rowild** [comment]
@rstoenescu Thank you very much for bringing that to my attention! Much appreciated! :-)
